source/common/http/conn_manager_impl.h

```diff
@@ -462,6 +469,18 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
     virtual Tracing::OperationName operationName() const override;
     virtual const std::vector<Http::LowerCaseString>& requestHeadersForTags() const override;
 
+    // Http::DownstreamWatermarkProvider
+    void addCallbacks(DownstreamWatermarkCallbacks& watermark_callbacks) override {
+      // This is currently expected to be called exactly once per stream, by the router filter.
```

Comment:
AFAIK there should be no problem for having multiple callbacks, so is this just for sanity checking? If so, I might be more clear in the comment that it should work but we are asserting just to get more eyes on the use case, etc.